### PR TITLE
Buildkite assumable ID example plus tweaks to current ID docs

### PR DESCRIPTION
**Pull request by SharpRake**
### What should this PR do?

Adds Matt's Buildktie example and reorganizes a bit of the ID docs to tidy things up.

Part of #671 , but does not resolve it.

### How should this PR be tested?

There's multiple ways to get this set up within buildkite. I tried to make things clear without being too prescriptive, hence I use their "Getting started" guide as the example to build off of. If someone is using buildkite already, they probably know enough to just add the `chainctl auth login` command to their pipeline and start using the identity.

All this to say, a tech test might be challenging so LMK if you need a hand.

### CODE REVIEW

File: content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+
+## Testing the identity with a Buildkite pipeline
+
+To test the identity you created with Terraform in the previous section, navigate to your Buildkite pipeline. From the Buildkite Dashboard, click **Pipelines** in the top navigation bar and then click on the pipeline you specified in the `buildkite.tf` file. 
+
+From there, click the **Edit Steps** button to add the following commands to a step in your pipeline.
+
+```
+- command: |
+	wget -O chainctl "https://dl.enforce.dev/chainctl/latest/chainctl_linux_$(uname -m)"
```
Comment (jamonation) [inline content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md:326]:
```suggestion
	curl -o chainctl "https://dl.enforce.dev/chainctl/latest/chainctl_$(uname -s | tr '[:upper:]' '[:lower:]')_$(uname -m)"
```

I think we should try this version instead? I ran into an issue with it not working on macOS with my buildkite runner.

File: content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+	chmod +x chainctl
+	mv chainctl /usr/bin
```
Comment (jamonation) [inline content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md:328]:
```suggestion
```

I don't think it needs to get mv'ed anywhere - better to keep it confined to the build directory and then run `./chainctl auth ...` in the next step.

File: content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+
+	# Assume
+	chainctl auth login --identity-token $(buildkite-agent oidc request-token --audience issuer.enforce.dev) --identity <your buildkite identity>
```
Comment (jamonation) [inline content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md:331]:
```suggestion
	./chainctl auth login --identity-token $(buildkite-agent oidc request-token --audience issuer.enforce.dev) --identity <your buildkite identity>
```

File: content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+```
+
+You could add the commands for testing the identity like this.
+
+```
+steps:
+  - label: "Buildkite test"
+	command:
+	- 'buildkite-agent pipeline upload'
+	- 'wget -O chainctl "https://dl.enforce.dev/chainctl/latest/chainctl_linux_$(uname -m)"'
```
Comment (jamonation) [inline content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md:352]: Depending on the changes to the steps before this, these commands will need to be edited too.

File: content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md [hunk 5]
```diff
@@ -0,0 +1,420 @@
...
+. . .
+
+Apply complete! Resources: 3 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+buildkite-identity = "<your buildkite identity>"
+```
+
+This is the identity's UIDP (unique identity path), which you configured the `buildkite.tf` file to emit in the previous section. Note this value down, as you'll need it when you test this identity using a Buildkite workflow. If you need to retrieve this UIDP later on, though, you can always run the following `chainctl` command to obtain a list of the UIDPs of all your existing identities.
```
Comment (jamonation) [inline content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md:309]:
```suggestion
This is the identity's [UIDP (unique identity path)](/chainguard/chainguard-enforce/reference/events/#uidp-identifiers), which you configured the `buildkite.tf` file to emit in the previous section. Note this value down, as you'll need it when you test this identity using a Buildkite workflow. If you need to retrieve this UIDP later on, though, you can always run the following `chainctl` command to obtain a list of the UIDPs of all your existing identities.
```

Comment (jamonation) [reply content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md:309]: Just adding this here for now, though I think I should split that UIDP thing into a separate reference page at some point

File: content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md [hunk 6]
```diff
@@ -0,0 +1,420 @@
...
+```sh
+terraform apply
+```
+
+Before going through with applying the Terraform configuration, this command will prompt you to confirm that you want it to do so. Enter `yes` to apply the configuration.
+
+```
+. . .
+
+Plan: 3 to add, 0 to change, 0 to destroy.
```
Comment (jamonation) [inline content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md:285]:
```suggestion
Plan: 4 to add, 0 to change, 0 to destroy.
```

Not sure where the diff is, but I get 4 to add here, not 3.

File: content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md [hunk 7]
```diff
@@ -0,0 +1,420 @@
...
+
+  Enter a value:
+```
+
+After pressing `ENTER`, the command will complete and will output an `buildkite-identity` value.
+
+```
+. . .
+
+Apply complete! Resources: 3 added, 0 changed, 0 destroyed.
```
Comment (jamonation) [inline content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md:302]: Now I run into this - I don't know if it is my Enforce user/group, or something with Terraform, or something else. Did you run into any issues like this one?

```
chainguard_group.user-group: Creating...
chainguard_group.user-group: Creation complete after 0s [id=fcf00e0c8f6a79958403f31a833daee115d63402]
chainguard_policy.cgr-trusted: Creating...
chainguard_identity.buildkite: Creating...
╷
│ Error: rpc error: code = PermissionDenied desc = caller does not have the required capabilities at "fcf00e0c8f6a79958403f31a833daee115d63402", missing: identity.create
│
│   with chainguard_identity.buildkite,
│   on buildkite.tf line 1, in resource "chainguard_identity" "buildkite":
│    1: resource "chainguard_identity" "buildkite" {
│
╵
╷
│ Error: rpc error: code = PermissionDenied desc = caller does not have the required capabilities at "fcf00e0c8f6a79958403f31a833daee115d63402", missing: policy.create
│
│   with chainguard_policy.cgr-trusted,
│   on sample.tf line 10, in resource "chainguard_policy" "cgr-trusted":
│   10: resource "chainguard_policy" "cgr-trusted" {
│
╵
```

**Review by jamonation — APPROVED** [review]
So satisfying to see a green buildkite build at the end! It took me a few tries, mainly around the terraform bits. Not sure if we should have a standalone doc about adding the provider, or make it a step in the prerequisites?

File: content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md (hunk 1 above)
Comment (SharpRake) [reply content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md:326]: I think all your suggestions to this part are going to normalize this process, thanks a ton Jamon. I was running into some of these same issues but was having a hard time pinning down clean solutions.

File: content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md (hunk 5 above)
Comment (SharpRake) [reply content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md:309]: thanks for this! I added it to the github actions example too

File: content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md (hunk 7 above)
Comment (SharpRake) [reply content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md:302]: yes! It's a weird quirk but between `terraform plan` and `terraform apply` you need to `chainctl auth login`. I found a thread about it on slack a while ago (can't find it today) but it's just something you have to do. Something about running `terraform plan` makes you logout of chainctl. I'll add the command here and to the actions guide

**Pull request merged**